envi/archs/arm/disasm.py

```diff
@@ -220,107 +229,123 @@ def p_dp_imm_shift(opval, va):
     return (opcode, dp_mnem[ocode], olist, iflags)
 
 # specialized mnemonics for p_misc
-qop_mnem = ('qadd','qsub','qdadd','qdsub')
-smla_mnem = ('smlabb','smlabt','smlatb','smlatt',)
-smlal_mnem = ('smlalbb','smlalbt','smlaltb','smlaltt',)
+qop_mnem = ('qadd','qsub','qdadd','qdsub') # used in misc1
+smla_mnem = ('smlabb','smlatb','smlabt','smlatt',)
+smlal_mnem = ('smlalbb','smlaltb','smlalbt','smlaltt',)
 smul_mnem = ('smulbb','smultb','smulbt','smultt',)
 smlaw_mnem = ('smlawb','smlawt',)
 smulw_mnem = ('smulwb','smulwt',)
 
 def p_misc(opval, va):  
     # 0x0f900000 = 0x01000000 or 0x01000010 (misc and misc1 are both parsed at the same time.  see the footnote [2] on dp instructions in the Atmel AT91SAM7 docs
-    if   opval & 0x0fc00000 == 0x01000000:
-        opcode = (IENC_MISC << 16) + 1
-        mnem = 'mrs'
-        r = (opval>>22) & 1
-        Rd = (opval>>12) & 0xf
-        olist = (
-            ArmRegOper(Rd, va=va),
-            ArmPgmStatRegOper(r),
-        )
-    elif opval & 0x0fb002f0 == 0x01200000:
+
+    #Including SBO and SBZ - rearranged for most exclusive to least
+    #updated reference names to match v7 reference ie Rm Rn Rd Rs m n etc
+    
+    #if opval & 0x0ff000f0 == 0x01200020:
+    if opval & 0x0FFFFFF0 == 0x012FFF20:  
+        opcode = (IENC_MISC << 16) + 5
+        mnem = 'bxj'
+        Rm = opval & 0xf
+        olist = ( ArmRegOper(Rm, va=va), )
+        
+    #elif opval & 0x0fb002f0 == 0x01200000:
+    elif opval & 0x0DB0F000 == 0x0120F000:
         opcode = (IENC_MISC << 16) + 2
         mnem = 'msr'        # register.   immediate has it's own parser in the 001 section
         r = (opval>>22) & 1
-        Rd = (opval) & 0xf
+        Rn = (opval) & 0xf
         olist = (
             ArmPgmStatRegOper(r),
-            ArmRegOper(Rd, va=va),
+            ArmRegOper(Rn, va=va),
         )
-    elif opval & 0x0ff000f0 == 0x01200020:
-        opcode = (IENC_MISC << 16) + 5
-        mnem = 'bxj'
-        Rm = opval & 0xf
-        olist = ( ArmRegOper(Rm, va=va), )
-        
-    elif opval & 0x0ff00090 == 0x01000080:
+
+    #smla 
+    #Mask and value are OK
+    elif opval & 0x0FF00090 == 0x01000080:
         opcode = (IENC_MISC << 16) + 9
-        xy = (opval>>5)&3
-        mnem = smla_mnem[xy]
+        mn = (opval>>5)&3
+        mnem = smla_mnem[mn]
         Rd = (opval>>16) & 0xf
-        Rn = (opval>>12) & 0xf 
-        Rs = (opval>>8) & 0xf
-        Rm = opval & 0xf
+        Ra = (opval>>12) & 0xf 
+        Rm = (opval>>8) & 0xf
+        Rn = opval & 0xf
         olist = (
             ArmRegOper(Rd, va=va),
-            ArmRegOper(Rm, va=va),
-            ArmRegOper(Rs, va=va),
             ArmRegOper(Rn, va=va),
+            ArmRegOper(Rm, va=va),
+            ArmRegOper(Ra, va=va),
         )
+    #smlaw
+    #mask and value are OK
     elif opval & 0x0ff000b0 == 0x01200080:
         opcode = (IENC_MISC << 16) + 10
-        y = (opval>>6)&1
-        mnem = smlaw_mnem[y]
+        m = (opval>>6)&1
```

Comment:
arrrgg! i hate it when they completely rewrite the docs.  hopefully fewer errors in the new docs than the old ones.